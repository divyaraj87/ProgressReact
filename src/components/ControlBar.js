import React, { Component} from 'react';
import propTypes from 'prop-types';
import {connect } from 'react-redux';
import { selectChange, progressChange } from '../actions';

export class ControlBar extends Component {
    render(){
        const {todos: { buttons, progressBar3 }, dispatchSelectChange, dispatchProgressChange} = this.props;
        return (
            <div class="controlContainer">        
               <select id="mySelect" onChange={() => dispatchSelectChange()}>
                    <option>progressBar1</option>
                    <option>progressBar2</option>
                    {progressBar3.completed && <option>progressBar3</option>}
               </select>
               <button                
                    onClick={() => dispatchProgressChange(buttons[0])}>+{buttons[0]}</button>
               <button                
                    onClick={() => dispatchProgressChange(buttons[1])}>+{buttons[1]}</button>
               <button                
                    onClick={() => dispatchProgressChange(buttons[2])}>{buttons[2]}</button>
                <button                
                    onClick={() => dispatchProgressChange(buttons[3])}>{buttons[3]}</button>
            </div>
        );
    }
}

ControlBar.prototypes = {
    todos: propTypes.object,
    dispatchSelectChange: propTypes.func,
    dispatchProgressChange: propTypes.func
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps, {
    dispatchSelectChange: selectChange,
    dispatchProgressChange: progressChange
})(ControlBar);