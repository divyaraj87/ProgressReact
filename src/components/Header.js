import React from 'react'

const Header = () => (
  <div>
    <h1>Progress Bar using React Redux</h1>
  </div>
)

export default Header
