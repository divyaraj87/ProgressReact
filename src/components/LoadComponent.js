import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { getExternalScript } from '../actions';

export class LoadComponent extends Component {
    render() {
        const { dispatchGetExternalScript } = this.props;
        return (
            <div class="LoadContainer">
                <button id="LoadButton" onClick={() => dispatchGetExternalScript()}>Click here to Load Progress Bar</button>
            </div>
        );
    }
}

LoadComponent.prototypes = {
   
    dispatchGetExternalScript: propTypes.func
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps,
    { dispatchGetExternalScript: getExternalScript })(LoadComponent);