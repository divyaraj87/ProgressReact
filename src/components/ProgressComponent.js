import React from 'react';

export const ProgressComponent = ({ bgcolor, completed, width}) => <div class="containerStyles" >
    <div style={{backgroundColor: `${bgcolor}`, width: `${width}%`, textAlign: 'right', height: '100%', borderRadius: 'inherit'}}>
        <span class="labelStyles">{`${completed}%`}</span>
    </div>
</div>