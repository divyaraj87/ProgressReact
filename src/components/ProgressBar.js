import React, { Component} from 'react';
import propTypes from 'prop-types';
import {connect } from 'react-redux';
import { ProgressComponent } from './ProgressComponent';
import { getExternalScript } from '../actions';

export class ProgressBar extends Component {
    render(){
        const { todos, dispatchGetExternalScript } = this.props;
        return (
            <div class="progressContainer">                
                <span>ProgressBar1</span>
                <ProgressComponent
                    bgcolor={todos.progressBar1.bgcolor}
                    width={todos.progressBar1.width}
                    completed={todos.progressBar1.completed}
                />
                <span>ProgressBar2</span>
                <ProgressComponent
                    bgcolor={todos.progressBar2.bgcolor}
                    width={todos.progressBar2.width}
                    completed={todos.progressBar2.completed}
                />
                {todos.progressBar3.completed && <span>ProgressBar3</span>}
                {todos.progressBar3.completed && <ProgressComponent
                    bgcolor={todos.progressBar3.bgcolor}
                    width={todos.progressBar3.width}
                    completed={todos.progressBar3.completed}
                />}

            </div>
        );
    }
    
}

ProgressBar.prototypes = {
    todos: propTypes.object,
    dispatchGetExternalScript: propTypes.func
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps,
    { dispatchGetExternalScript: getExternalScript})(ProgressBar);