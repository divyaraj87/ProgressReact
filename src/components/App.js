import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { getExternalScript } from '../actions';
import Header from './Header'
import ProgressBar from './ProgressBar'
import ControlBar from './ControlBar'
import LoadComponent from './LoadComponent'
import '../_style.scss';

export class App extends Component {
  render(){
      const { todos: { loadProgress, spinnerEnable }, dispatchGetExternalScript } = this.props;
      return (
        <div className="container">      
          {!loadProgress && <Header />}
          {!loadProgress && <LoadComponent /> }
          {spinnerEnable && <span>Loading...</span> }
          {loadProgress && <ProgressBar />}
          {loadProgress && <ControlBar />}
      </div>
      );
  }
 
}

App.prototypes = {
  todos: propTypes.object,
  dispatchGetExternalScript: propTypes.func
}

export const mapStateToProps = (state) => ({
  todos: state.todos
});

export default connect(mapStateToProps,
  { dispatchGetExternalScript: getExternalScript})(App);

