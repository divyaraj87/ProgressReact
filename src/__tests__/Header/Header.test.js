import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import  configureMockStore from 'redux-mock-store';
import { Header }  from '../../components/Header';

const dispatchSelectChangeSpy = sinon.spy();
const dispatchProgressChangeSpy = sinon.spy();
let mockProps = {
    bgcolor: '#000',
    completed: 10,
    width: 10
};
const mockStore = configureMockStore();
const store = mockStore({});
describe('Header', () => {
  let pageWrapper;

  describe('Header Page', () => {
    const originalProps = { ...mockProps };

    beforeEach(() => {
      pageWrapper = mount(       
        <Header />       
      );
    });

    afterEach(() => {
      mockProps = { ...originalProps };
    });

    it('should render Header', () => {
      expect(pageWrapper).toMatchSnapshot();
    });
  });
});
