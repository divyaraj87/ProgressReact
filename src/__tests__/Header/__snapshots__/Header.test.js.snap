// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Page should render Header 1`] = `undefined`;
